src/create-client.ts

```diff
@@ -212,11 +221,29 @@ export async function createClient(
     const lock = new Lock();
     try {
       _authkitClientState = "AUTHENTICATING";
+
+      if (organizationId) {
+        sessionStorage.setItem(
+          ORGANIZATION_ID_SESSION_STORAGE_KEY,
+          organizationId,
+        );
+      } else {
+        const accessToken = _getAccessToken();
+        if (accessToken) {
+          organizationId = getClaims(accessToken)?.org_id;
+        } else {
+          organizationId =
+            sessionStorage.getItem(ORGANIZATION_ID_SESSION_STORAGE_KEY) ??
+            undefined;
+        }
+      }
+
       if (await lock.acquireLock(REFRESH_LOCK)) {
```

Comment:
i can't really envision a case where it matters, but i wonder if we should move the organizationId determination into this part of the code that is already locked?

Reply:
That sounds right to me! Will make that change.